src/repository/repository.action.js

```diff
@@ -139,6 +143,89 @@ export const getIssues = url => {
   };
 };
 
+export const getCommits = url => {
+  return (dispatch, getState) => {
+    const accessToken = getState().auth.accessToken;
+
+    dispatch({ type: GET_REPOSITORY_COMMITS.PENDING });
+
+    fetchUrl(url, accessToken)
+      .then(data => {
+        dispatch({
+          type: GET_REPOSITORY_COMMITS.SUCCESS,
+          payload: data,
+        });
+      })
+      .catch(error => {
+        dispatch({
+          type: GET_REPOSITORY_COMMITS.ERROR,
+          payload: error,
+        });
+      });
+  };
+};
+
+const getCommitFromUrl = url => {
+  return (dispatch, getState) => {
+    const accessToken = getState().auth.accessToken;
+
+    dispatch({ type: GET_COMMIT.PENDING });
+
+    fetchUrl(url, accessToken)
+      .then(data => {
+        dispatch({
+          type: GET_COMMIT.SUCCESS,
+          payload: data,
+        });
+      })
+      .catch(error => {
+        dispatch({
+          type: GET_COMMIT.ERROR,
+          payload: error,
+        });
+      });
+  };
+};
+
+const getCommitDiffFromUrl = url => {
+  return (dispatch, getState) => {
+    const accessToken = getState().auth.accessToken;
+
+    dispatch({ type: GET_COMMIT_DIFF.PENDING });
+
+    fetchDiff(url, accessToken)
+      .then(data => {
+        dispatch({
+          type: GET_COMMIT_DIFF.SUCCESS,
+          payload: data,
+        });
+      })
+      .catch(error => {
+        dispatch({
+          type: GET_COMMIT_DIFF.ERROR,
+          payload: error,
+        });
+      });
+  };
+};
+
+export const getCommitDetails = commit => {
+  // eslint-disable-next-line no-unused-vars
+  return (dispatch, getState) => {
```

Comment:
You can omit `getState` here.

Reply:
Thanks @alejandronanez for your review 😃